### PR TITLE
[REF] mail: chatWindow, review and simplify scss

### PR DESCRIPTION
**Pull request by edi-odoo**
Part of the overall v16 SCSS optimization/restyle, task-2704984

task-2784763

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/src/components/chat_window/chat_window.scss [hunk 1]
```diff
@@ -23,44 +18,20 @@
...
+.o_ChatWindow_newMessageFormLabel,
 .o_ChatWindow_thread {
-    flex: 1 1 auto;
+    flex-basis: auto;
 }
 
+
 // ------------------------------------------------------------------
 // Style
 // ------------------------------------------------------------------
```
Comment (stefanorigano) [inline addons/mail/static/src/components/chat_window/chat_window.scss:37]: https://github.com/odoo/odoo/pull/85839/files#diff-e5ea01996c5bd038041497470854a9c1cf33913cce0b4b536cb783f6696c160aL41

Can't we use `border-lg-top` ? I know that the resulting value will not be exactly 6px but, actually, keeping these values consistent across the ui would be good.

---

https://github.com/odoo/odoo/pull/85839/files#diff-e5ea01996c5bd038041497470854a9c1cf33913cce0b4b536cb783f6696c160aR47

You can conbine the selectors, so 

```scss
&.o-focused:not(.o-mobile)
```

---

https://github.com/odoo/odoo/pull/85839/files#diff-e5ea01996c5bd038041497470854a9c1cf33913cce0b4b536cb783f6696c160aR53

`border-0` ?

File: addons/mail/static/src/components/chat_window/chat_window.scss [hunk 2]
```diff
@@ -83,13 +54,8 @@
...
-.o_ChatWindow_header {
-    border-radius: 3px 3px 0 0;
-}
-
 .o_ChatWindow_newMessageFormInput {
     outline: none;
-    border: $border-width solid gray('300'); // cancel firefox border on input focus
 }
 
 .o_ChatWindow_thread .o_ThreadView_messageList {
```
Comment (stefanorigano) [inline addons/mail/static/src/components/chat_window/chat_window.scss:61]: It sounds a bit off to have font-size for `MessageList` defined here :/
If it's a contextual value it should be probably be set using CSS variables instead.
@seb-odoo , any feedback on this?

**Review by stefanorigano — CHANGES_REQUESTED** [review]
@edi-odoo , some remarks for you.

The rest is fine for me, you can remove the draft state and wait for @seb-odoo review.

Thanks!

File: addons/mail/static/src/components/chat_window/chat_window.scss (hunk 2 above)
Comment (seb-odoo) [reply addons/mail/static/src/components/chat_window/chat_window.scss:61]: Indeed it's probably there for historical reasons. We should remove the specificity and add a conditional class in message list itself.

Comment (seb-odoo) [reply addons/mail/static/src/components/chat_window/chat_window.scss:61]: Note: no need to do all changes here. This should be done in either a preparation PR, or follow up PR, depending on which is easier.

File: addons/mail/static/src/components/chat_window/chat_window.scss [hunk 3]
```diff
@@ -23,44 +18,20 @@
...
-}
-
-.o_ChatWindow_newMessageFormLabel {
-    margin-right: map-get($spacers, 2);
-    flex: 0 0 auto;
-}
-
+.o_ChatWindow_header, 
+.o_ChatWindow_newMessageFormInput,
+.o_ChatWindow_newMessageFormLabel,
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/mail/static/src/components/chat_window/chat_window.scss:29]: Indeed, let me know when the PR is ready for you and I'll do the final check after these comments are done.

[the remark this answers is not in the document]
Comment (edi-odoo) [reply addons/mail/static/src/components/chat_window/chat_window.scss:28]: Didn't find a workaround for this. I've split the selectors :)

File: addons/mail/static/src/components/chat_window/chat_window.scss (hunk 1 above)
Comment (edi-odoo) [reply addons/mail/static/src/components/chat_window/chat_window.scss:36]: For `.border-0`; `.o_Composer` is a different component altogether, child of `o_ThreadView/o_ChatWindow_thread`

File: addons/mail/static/src/components/chat_window/chat_window.scss [hunk 4]
```diff
@@ -23,42 +17,26 @@
...
-    margin-top: map-get($spacers, 1);
-    display: flex;
-    align-items: center;
-}
+    flex-basis: auto;
+} 
 
 .o_ChatWindow_newMessageFormInput {
-    flex: 1 1 auto;
+    flex-basis: auto;
```
Comment (seb-odoo) [inline addons/mail/static/src/components/chat_window/chat_window.scss:31]: Aren't all of those the default value, so could be removed from the file?

Comment (edi-odoo) [reply addons/mail/static/src/components/chat_window/chat_window.scss:31]: They should indeed, although when I tested it without, it didn't render properly. I'll test it again to make sure.

Comment (edi-odoo) [reply addons/mail/static/src/components/chat_window/chat_window.scss]: Tested and I can't seem to recreate my previous issue, I've removed them :)

**Comment by edi-odoo** [comment]
@seb-odoo to be merged 🙏

**Comment by stefanorigano** [comment]
Hello @odoo/rd-discuss , could you review? 
Thanks

**Comment by stefanorigano** [comment]
Hello @odoo/rd-discuss , could you review? 
Thanks

File: addons/mail/static/src/components/chat_window/chat_window.xml [hunk 5]
```diff
@@ -3,17 +3,18 @@
...
+            <div class="o_ChatWindow position-absolute bottom-0 d-flex flex-column rounded-lg-top bg-white" t-attf-class="{{ className }}" tabindex="0" t-att-data-visible-index="chatWindow ? chatWindow.visibleIndex : undefined"
+            t-att-class="{
+                'o-focused': chatWindow and chatWindow.isFocused,
+                'o-folded': chatWindow and chatWindow.isFolded,
+                'o-fullscreen w-100 h-100': props.isFullscreen,
+                'o-mobile position-fixed': messaging and messaging.device.isMobile,
+                'mw-100 mh-100': !messaging and !messaging.device.isMobile,
+                'o-new-message': chatWindow and !chatWindow.thread,
+            }" t-att-style="chatWindow ? chatWindow.componentStyle : undefined" t-on-keydown="_onKeydown" t-on-focusout="_onFocusout" t-att-data-chat-window-local-id="chatWindow ? chatWindow.localId : undefined" t-att-data-thread-local-id="chatWindow ? (chatWindow.thread ? chatWindow.thread.localId : '') : undefined" t-ref="root"
+        >
```
Comment (alexkuhn) [inline addons/mail/static/src/components/chat_window/chat_window.xml:15]: Unnecessary diff from rebase: should no longer guard `chatWindow`.
The diff should only have change in `class` attribute.

**Review by alexkuhn — CHANGES_REQUESTED** [review]
Looks good otherwise

File: addons/mail/static/src/components/chat_window/chat_window.xml (hunk 5 above)
Comment (edi-odoo) [reply addons/mail/static/src/components/chat_window/chat_window.xml]: Thanks, could you check again please ?

File: addons/mail/static/src/components/chat_window/chat_window.xml [hunk 6]
```diff
@@ -3,17 +3,17 @@
...
-                    'o-mobile': messaging.device.isMobile,
-                    'o-new-message': !chatWindow.thread,
-                }" t-att-style="chatWindow.componentStyle" t-on-keydown="_onKeydown" t-on-focusout="_onFocusout" t-att-data-chat-window-local-id="chatWindow.localId" t-att-data-thread-local-id="chatWindow.thread ? chatWindow.thread.localId : ''" t-ref="root"
+            <div class="o_ChatWindow position-absolute bottom-0 d-flex flex-column rounded-lg-top bg-white" t-attf-class="{{ className }}" tabindex="0" t-att-data-visible-index="chatWindow.visibleIndex" t-att-class="{
+                'o-focused': chatWindow and chatWindow.isFocused,
+                'o-folded': chatWindow and chatWindow.isFolded,
+                'o-fullscreen w-100 h-100': props.isFullscreen,
+                'o-mobile position-fixed': messaging and messaging.device.isMobile,
+                'mw-100 mh-100': !messaging and !messaging.device.isMobile,
+                'o-new-message': chatWindow and !chatWindow.thread,
```
Comment (alexkuhn) [inline addons/mail/static/src/components/chat_window/chat_window.xml:12]: Still some unnecessary guards.
- No need to `chatWindow and` (it's covered by top-level `t-if="chatWindow"`)
- No need to `messaging and` (it's also implicitly covered by top-level `t-if="chatWindow"`)

File: addons/mail/static/src/components/chat_window/chat_window.xml [hunk 7]
```diff
@@ -3,17 +3,17 @@
...
         <t t-if="chatWindow">
-            <div class="o_ChatWindow bg-white" t-attf-class="{{ className }}" tabindex="0" t-att-data-visible-index="chatWindow.visibleIndex"
-                t-att-class="{
-                    'o-focused': chatWindow.isFocused,
-                    'o-folded': chatWindow.isFolded,
-                    'o-fullscreen': props.isFullscreen,
-                    'o-mobile': messaging.device.isMobile,
-                    'o-new-message': !chatWindow.thread,
-                }" t-att-style="chatWindow.componentStyle" t-on-keydown="_onKeydown" t-on-focusout="_onFocusout" t-att-data-chat-window-local-id="chatWindow.localId" t-att-data-thread-local-id="chatWindow.thread ? chatWindow.thread.localId : ''" t-ref="root"
+            <div class="o_ChatWindow position-absolute bottom-0 d-flex flex-column rounded-lg-top bg-white" t-attf-class="{{ className }}" tabindex="0" t-att-data-visible-index="chatWindow.visibleIndex" t-att-class="{
```
Comment (alexkuhn) [inline addons/mail/static/src/components/chat_window/chat_window.xml:6]: no need to move `t-att-class` inline, this is unnecessary diff.

File: addons/mail/static/src/components/chat_window/chat_window.xml [hunk 8]
```diff
@@ -3,17 +3,17 @@
...
+                'o-focused': chatWindow and chatWindow.isFocused,
+                'o-folded': chatWindow and chatWindow.isFolded,
+                'o-fullscreen w-100 h-100': props.isFullscreen,
+                'o-mobile position-fixed': messaging and messaging.device.isMobile,
+                'mw-100 mh-100': !messaging and !messaging.device.isMobile,
```
Comment (alexkuhn) [inline addons/mail/static/src/components/chat_window/chat_window.xml:11]: `!messaging` is always `false`, so this style is never applied.

File: addons/mail/static/src/components/chat_window/chat_window.xml [hunk 9]
```diff
@@ -3,17 +3,18 @@
 
     <t t-name="mail.ChatWindow" owl="1">
         <t t-if="chatWindow">
-            <div class="o_ChatWindow bg-white" t-attf-class="{{ className }}" tabindex="0" t-att-data-visible-index="chatWindow.visibleIndex"
+            <div class="o_ChatWindow position-absolute bottom-0 d-flex flex-column rounded-lg-top bg-white" t-attf-class="{{ className }}" tabindex="0" t-att-data-visible-index="chatWindow.visibleIndex"
```
Comment (alexkuhn) [inline addons/mail/static/src/components/chat_window/chat_window.xml:6]: just remove the trailing space at the end of line

**Review by alexkuhn — CHANGES_REQUESTED** [review]
LGTM, just small trailing space in diff to remove. Thanks!

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Comment by seb-odoo** [comment]
@robodoo retry
